### PR TITLE
Attempt to refactor use of FLAP.

### PR DESCRIPTION
**Pull request by tclune**
- Migrated CapOptions into ./gridcomps/cap
 - Reduced use of `#ifdef`

## Description

- Relocated 2 files from `./base` to `./gridcomps/Cap`
- Improved separation of CapOptions and FlapCLI layers.
- 
## Motivation and Context
Cleanup.

## How Has This Been Tested?
Just MAPL tests.  Further testing with GCM is warranted.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by Gvilla1000-nasa — APPROVED** [review]

**Comment by Gvilla1000-nasa** [comment]
I don't see any changes here impacting the other refactoring PR, but I blocked it until the other one gets resolved.

**Review by mathomp4 — APPROVED** [review]
Approved for CMake

**Comment by mathomp4** [comment]
Oh wow. GNU does not like this. I wonder if it's a bug in the compiler?

**Comment by mathomp4** [comment]
I'll attach a `make.log` here for your perusal:
[make.log](https://github.com/GEOS-ESM/MAPL/files/7121950/make.log)

File: gridcomps/Cap/FlapCLI.F90 [hunk 1]
```diff
@@ -5,47 +5,55 @@ module MAPL_FlapCLIMod
    use MPI
    use ESMF
    use FLAP
-   use MAPL_KeywordEnforcerMod
-   use MAPL_ExceptionHandling
+   use mapl_KeywordEnforcerMod
+   use mapl_ExceptionHandling
+   use mapl_CapOptionsMod, only: MAPL_FlapCli => MAPL_CapOptions
```
Comment (Gvilla1000-nasa) [inline gridcomps/Cap/FlapCLI.F90:10]: Seems like the use of the name `MAPL_FlapCli` to rename the `MAPL_CapOptions` interface for the `new_CapOptions` procedure and still using it in this module as the name of the interface holding the `new_CapOptions_from_flap` procedure is causing conflicts. Commenting this line though makes it fail in the external components like Fvdycore.

**Comment by Gvilla1000-nasa** [comment]
With the result from `new_CapOptions_from_flap` being now of type `MAPL_CapOptions` instead of `MAPL_FlapCLI`, wouldn't lines 21-22 of `Tests/ExtDataDriver.F90` need to be modified from:

```fortran
  cli = MAPL_FlapCLI(description='extdata driver',authors='gmao')
  cap_options=MAPL_CapOptions(cli)
```

to

```fortran
 cap_options = MAPL_FlapCLI(description='extdata driver',authors='gmao')
```

and remove the `type(MAPL_FlapCLI) :: cli`?

**Comment by tclune** [comment]
The intent was to avoid needing to edit the application programs which reside in other repos.  (I.e., to preserve backwards compatibility.    The second line is supposed to just to a copy now which makes it "silly" in the grand scheme of things, but lets those programs be unchanged.    I'll try to poke around soon to see if this is a gfortran bug and/or if there is a workaround.

**Pull request merged**